src/main/kotlin/app/revanced/patcher/util/xml/LazyXMLInputSource.kt

```diff
@@ -0,0 +1,61 @@
+package app.revanced.patcher.util.xml
+
+import app.revanced.patcher.apk.Apk
+import app.revanced.patcher.resource.boolean
+import com.reandroid.apk.xmlencoder.EncodeException
+import com.reandroid.apk.xmlencoder.XMLEncodeSource
+import com.reandroid.arsc.chunk.xml.ResXmlDocument
+import com.reandroid.xml.XMLDocument
+import com.reandroid.xml.XMLElement
+import com.reandroid.xml.source.XMLDocumentSource
+
+/**
+ * Archive input source that lazily encodes the [XMLDocument] when you read from it.
+ *
+ * @param name The file name of this input source.
+ * @param document The [XMLDocument] to encode.
+ * @param resources The [Apk.ResourceContainer] to use for encoding.
+ */
+internal class LazyXMLInputSource(
```

Comment:
This class does not belong in ReVanced Patcher because it is strictly dealing with XML files from arsclib. Please create a separate module that wraps around `arsclib` and use it in ReVanced Patcher.